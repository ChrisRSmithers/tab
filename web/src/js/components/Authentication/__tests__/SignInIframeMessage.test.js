/* eslint-env jest */

import React from 'react'
import {
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import Typography from '@material-ui/core/Typography'
import {
  getUrlParameters
} from 'js/utils/utils'

jest.mock('js/utils/utils')

describe('SignInIframeMessage tests', () => {
  it('renders without error', () => {
    const SignInIframeMessage = require('js/components/Authentication/SignInIframeMessage').default
    shallow(
      <SignInIframeMessage />
    )
  })

  it('redirects the page when clicking the sign-in button', () => {
    const SignInIframeMessage = require('js/components/Authentication/SignInIframeMessage').default

    // Mock window.open
    window.open = jest.fn()

    const wrapper = shallow(
      <SignInIframeMessage />
    )
    const signInButton = wrapper.find('[data-test-id="sign-in-iframe-message-button"]')

    expect(window.open).not.toHaveBeenCalled()
    signInButton.simulate('click')
    expect(window.open).toHaveBeenCalled()
  })

  it('has the expected copy', () => {
    const SignInIframeMessage = require('js/components/Authentication/SignInIframeMessage').default
    const wrapper = shallow(
      <SignInIframeMessage />
    )
    expect(
      wrapper.find(Typography)
        .first().children().text())
      .toBe(`Let's get started!`)
    expect(
      wrapper.find(Typography)
        .at(1).children().text())
      .toBe(
        `Sign in to customize your new tab page and raise money for your favorite causes.`)
  })

  it('shows the explanation copy when an anonymous user is now required to sign in', () => {
    getUrlParameters.mockReturnValueOnce({
      mandatory: 'true'
    })
    const SignInIframeMessage = require('js/components/Authentication/SignInIframeMessage').default
    const wrapper = shallow(
      <SignInIframeMessage />
    )
    expect(
      wrapper.find(Typography)
        .first().children().text())
      .toBe(`Great job so far!`)
    expect(
      wrapper.find(Typography)
        .at(1).children().text())
      .toMatch(
        `You've already made a positive impact! Let's keep this progress safe:`)
  })

  it('matches expected snapshot', () => {
    const SignInIframeMessage = require('js/components/Authentication/SignInIframeMessage').default
    const wrapper = shallow(
      <SignInIframeMessage />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
