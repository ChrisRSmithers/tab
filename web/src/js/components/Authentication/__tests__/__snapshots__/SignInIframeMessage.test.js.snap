// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInIframeMessage tests matches expected snapshot 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "backgroundColor": "#FFF",
      "maxWidth": 400,
      "padding": 24,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <h3>
    Let's get started!
  </h3>
  <p>
    Sign in to customize your new tab page and raise money for your favorite causes.
  </p>
  <span
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-end",
        "marginTop": 24,
      }
    }
  >
    <RaisedButton
      data-test-id="sign-in-iframe-message-button"
      disabled={false}
      fullWidth={false}
      label="SIGN IN"
      labelPosition="after"
      onClick={[Function]}
      primary={true}
      secondary={false}
    />
  </span>
</Paper>
`;
